app/controllers/projects_controller.rb

```diff
@@ -153,10 +147,34 @@ def archived
     @projects = Project.archived
   end
 
+  def reports
+    since = params[:since].nil? ? nil : params[:since].to_i.months.ago
+    @service = IterationService.new(@project, since)
+    current_month = (@service.current_iteration_number / 4).floor
+    @current_iteration_start_date = @service.date_for_iteration_number(@service.current_iteration_number).to_date.to_s(:short)
+    @current_iteration_end_date   = (@service.date_for_iteration_number(@service.current_iteration_number + 1) - 1.day).to_date.to_s(:short)
+    @current_iteration            = @service.iteration_details(@service.backlog_iterations.first)
+    @accepted_points              = @service.current_iteration_details['accepted']
+    @accepted_rate                = @service.current_iteration_details['accepted'] / @current_iteration[:points]
+    @last_iteration               = @service.backlog_iterations.last
+
```

Comment:
sorry my lack of context here, but this is a classic example which a presenter would fit better.. something like:
```ruby
report = ReportPresenter.new(@project, params[:since])
report.current_iteration_start_date => ...
report.last_iteration => ...
```
and so on.. you can also expose it as `@report` and use it in the view